client/src/modules/user/AuthBox/components/SignInForm.jsx

```diff
@@ -130,21 +131,19 @@ class SignInForm extends PureComponent {
     this.setState({ pending: true });
     this.pendingPromise = makeAbortablePromise(signIn(email, password));
 
-    return this.pendingPromise.promise
-      .then(() => this.setState({ pending: false }))
-      .catch(err => {
-        if (!(err instanceof AbortPromiseException)) {
-          const newState = { pending: false };
-
-          if (err instanceof UnauthorizedException) {
-            newState.signInErrorId = 'user.actions.sign-in.invalid-credentials';
-          } else {
-            newState.signInErrorId = 'common.something-went-wrong';
-          }
+    return this.pendingPromise.promise.catch(err => {
+      if (!(err instanceof AbortPromiseException)) {
+        const newState = { pending: false };
 
-          this.setState(newState);
+        if (err instanceof UnauthorizedException) {
+          newState.signInErrorId = 'user.actions.sign-in.invalid-credentials';
+        } else {
+          newState.signInErrorId = 'common.something-went-wrong';
```

Comment:
you don't need to nest it here. 

just put it next to each other, also else is not necessary. Cause if the error is not an instance of AbortPromiseException and UnauthoriedException, which means it is ANY other error we need to handle this someway. 

```
 if (!(err instanceof AbortPromiseException)) {
    //....
  }

 if (err instanceof UnauthorizedException) {
   //....
 }

 newState.signInErrorId = 'common.something-went-wrong';
```

Reply:
This is old logic, only first then was removed... You're right - else is not necessary. But all changes that affect state should be performed  only when an error is not `instanceof AbortPromiseException`. So Id leaves it here. If your sign action fails and you click cancel before this async returns the result you'll get an error: can't change setState on unmounted component...